packages/api-folders-so-ddb/src/index.ts

```diff
@@ -0,0 +1,303 @@
+        name
+    });
+};
+
+export const createStorageOperations = (params: FoldersStorageParams): FoldersStorageOperations => {
+    const { table: tableName, documentClient, attributes } = params;
+    if (attributes) {
+        Object.values(attributes).forEach(attrs => {
+            Object.keys(attrs).forEach(isReserved);
+        });
+    }
+
+    const table = createTable({ table: tableName, documentClient });
+
+    const entities = {
+        folders: createFolderEntity(table, attributes ? attributes[ENTITIES.FOLDER] : {}),
+        entries: createEntryEntity(table, attributes ? attributes[ENTITIES.ENTRY] : {})
+    };
+
+    const createFolderKeys = ({
+        id,
+        tenant,
+        locale
+    }: Pick<Folder, "id" | "tenant" | "locale">) => ({
+        PK: `T#${tenant}#L#${locale}#FOLDER#${id}`,
+        SK: `A`
+    });
+
+    const createFolderGsiKeys = ({
+        tenant,
+        locale,
+        category,
+        slug
+    }: Pick<Folder, "tenant" | "locale" | "category" | "slug">) => ({
+        GSI1_PK: `T#${tenant}#L#${locale}#CATEGORY#${category}#FOLDERS`,
+        GSI1_SK: slug
+    });
+
+    const createEntryKeys = ({ id, tenant, locale }: Pick<Entry, "id" | "tenant" | "locale">) => ({
+        PK: `T#${tenant}#L#${locale}#ENTRY#${id}`,
+        SK: `A`
+    });
+
+    const createEntryGsiKeys = ({
+        tenant,
+        locale,
+        folderId,
+        id
+    }: Pick<Entry, "tenant" | "locale" | "folderId" | "id">) => ({
+        GSI1_PK: `T#${tenant}#L#${locale}#FOLDER#${folderId}#ENTRIES`,
+        GSI1_SK: id
+    });
+
+    return {
+        async createFolder({ folder }): Promise<Folder> {
+            const keys = {
+                ...createFolderKeys(folder),
+                ...createFolderGsiKeys(folder)
+            };
+
+            try {
+                await entities.folders.put({
+                    ...cleanupItem(entities.folders, folder),
+                    TYPE: "folder",
+                    ...keys
+                });
+                return folder;
+            } catch (error) {
+                throw WebinyError.from(error, {
+                    message: "Could not create folder.",
+                    code: "CREATE_FOLDER_ERROR",
+                    data: { keys }
+                });
+            }
+        },
+
+        async getFolder({ tenant, locale, id, slug, category }): Promise<Folder> {
+            try {
+                let result;
+                if (id) {
+                    const response = await entities.folders.get(
+                        createFolderKeys({ id, tenant, locale })
+                    );
+                    if (response.Item) {
+                        result = response.Item;
+                    }
+                } else if (slug) {
+                    result = await queryOne({
+                        entity: entities.folders,
+                        partitionKey: `T#${tenant}#L#${locale}#CATEGORY#${category}#FOLDERS`,
```

Comment:
Maybe extract this into a utility function? This string is constructed 3 times in this single file. And for Entries as well.